recursion/src/recursive_challenger.rs

```diff
@@ -34,6 +35,35 @@ pub trait RecursiveChallenger<F: Field> {
         (0..count).map(|_| self.sample(circuit)).collect()
     }
 
+    fn sample_bits(
+        &mut self,
+        circuit: &mut CircuitBuilder<F>,
+        total_num_bits: usize,
+        num_bits: usize,
+    ) -> Vec<Target> {
+        let x = self.sample(circuit);
+        // Adds one public input per bit (there are as many bits as for a field element).
```

Comment:
perhaps to signal the bits are expected to be public inputs
```suggestion
    fn sample_public_bits(
        &mut self,
        circuit: &mut CircuitBuilder<F>,
        total_num_bits: usize,
        num_bits: usize,
    ) -> Vec<Target> {
        let x = self.sample(circuit);
        // Adds one public input per bit (there are as many bits as for a field element).
```